### PR TITLE
macOS bash 5 & zsh doesn't support shopt -s inherit_errexit

### PR DESCRIPTION
**Pull request by NorseGaud**
# Description

macOS/ZSH doesn't support shopt -s inherit_errexit

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Improvement (non-breaking change that does improve existing functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation (README.md or the documentation under `docs/`)
- [x] If necessary I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by casperklein** [comment]
I am wondering why you use ZSH with BASH scripts, when you encounter problems. Using BASH instead would be the obvious solution.

**Comment by NorseGaud** [comment]
It's default for macOS

**Comment by casperklein** [comment]
But BASH is available and there is nothing preventing you to use it? (just curious, not using MacOS myself).

**Comment by casperklein** [comment]
Okay, afaik that feature was introduced in BASH 4.4

**Comment by casperklein** [comment]
Not directly related to this At the best, linting/building etc. should take place in a container. That way, we could get rid of all OS/Shell version specific adjustments.

File: test/linting/lint.sh [hunk 1]
```diff
@@ -105,7 +104,7 @@ function _shellcheck
     -not -path './target/docker-configomat/*'
   )"
   # macOS lacks parity for `-executable` but presently produces the same results: https://stackoverflow.com/a/4458361
-  [[ "$(uname)" == "Darwin" ]] && FIND_EXEC="-perm +111 -type l -or" || FIND_EXEC="-executable"
+  [[ "$(uname)" == "Darwin" ]] && FIND_EXEC="-perm -711" || FIND_EXEC="-executable"
```
Comment (casperklein) [inline test/linting/lint.sh:107]: This change isn't mentioned in the PR description. Can you add a short description for future reference?

**Review by casperklein — APPROVED** [review]

**Comment by casperklein** [comment]
```
bash-5.1$ make lint
./test/linting/lint.sh: line 22: shopt: inherit_errexit: invalid shell option name
```

I just recognized, that bash 5.1 > 4.4 😆 So the feature should be available.

What happens, when you run `shopt -s inherit_errexit` directly in your bash shell?

**Comment by NorseGaud** [comment]
Hi @casperklein, I figured it out. In our contributor guide, I see that we're telling users to install brew's bash (5.x). This is installed into /usr/local/bin and works when doing `bash --version`. However, Apple still has `/bin/bash` available and that's version 3.x. 

This is because of `#! /bin/bash` in the top of the lint script. We should use `#!/usr/bin/env bash` instead so I'll PR for that.

**Comment by casperklein** [comment]
> Not directly related to this: At the best, linting/building etc. should take place in a container. That way, we could get rid of all OS/Shell version specific adjustments.

For everyone interested in this, I hacked something together. Take a look at https://github.com/docker-mailserver/docker-mailserver/issues/2449 and give feedback, if that works for you.

**Pull request closed without merging**